ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -6,6 +6,16 @@ set -o pipefail
 
 trap 'FRC=$?; createUpgradeJunit; debug' EXIT TERM
 
+export HOME="${HOME:-/tmp/home}"
+export XDG_RUNTIME_DIR="${HOME}/run"
+export REGISTRY_AUTH_PREFERENCE=podman # TODO: remove later, used for migrating oc from docker to podman
+mkdir -p "${XDG_RUNTIME_DIR}"
+# After cluster is set up, ci-operator make KUBECONFIG pointing to the installed cluster,
+# to make "oc registry login" interact with the build farm, set KUBECONFIG to empty,
+# so that the credentials of the build farm registry can be saved in docker client config file.
```

Comment:
does only `cucushift-chainupgrade-toimage` need save build-farm's credentials? and i wonder why chain upgrade did not hit such issue before and even other profiles?

Reply:
actually I also have the same question, from my testing, sounds build05 has some different auth configurations.
```
---> The following commands are executed in a running pod on build05 build farm
---> accessing a build04 image, succeed.
$ env "NO_PROXY=*" "no_proxy=*" /cli/oc adm release info registry.build04.ci.openshift.org/ci-op-7hj98456/release@sha256:f5eaf0248779a0478cfd83f055d56dc7d755937800a68ad55f6047c503977c44 --loglevel=8
....
I0117 09:47:02.282915      60 round_trippers.go:463] GET https://registry.build04.ci.openshift.org/openshift/token?scope=repository%3Aci-op-7hj98456%2Frelease%3Apull
I0117 09:47:02.282924      60 round_trippers.go:469] Request Headers:
I0117 09:47:02.282929      60 round_trippers.go:473]     User-Agent: oc/4.10.0 (linux/amd64) kubernetes/5f40be4
I0117 09:47:02.304445      60 round_trippers.go:574] Response Status: 200 OK in 21 milliseconds
I0117 09:47:02.304463      60 round_trippers.go:577] Response Headers:
I0117 09:47:02.304467      60 round_trippers.go:580]     Content-Type: application/json
I0117 09:47:02.304471      60 round_trippers.go:580]     Docker-Distribution-Api-Version: registry/2.0
I0117 09:47:02.304474      60 round_trippers.go:580]     Content-Length: 49
I0117 09:47:02.304477      60 round_trippers.go:580]     Date: Wed, 17 Jan 2024 09:47:02 GMT
I0117 09:47:02.304480      60 round_trippers.go:580]     Set-Cookie: bfb46326e94010678bafb873b237ccc4=1ab2c7fe098d6faa9b8fde83b74da505; path=/; HttpOnly; Secure; SameSite=None
I0117 09:47:02.304536      60 round_trippers.go:463] GET https://registry.build04.ci.openshift.org/v2/ci-op-7hj98456/release/manifests/sha256:f5eaf0248779a0478cfd83f055d56dc7d755937800a68ad55f6047c503977c44
I0117 09:47:02.304541      60 round_trippers.go:469] Request Headers:
I0117 09:47:02.304544      60 round_trippers.go:473]     Accept: application/vnd.docker.distribution.manifest.list.v2+json
I0117 09:47:02.304548      60 round_trippers.go:473]     Accept: application/vnd.docker.distribution.manifest.v2+json
I0117 09:47:02.304551      60 round_trippers.go:473]     Accept: application/vnd.oci.image.manifest.v1+json
I0117 09:47:02.304555      60 round_trippers.go:473]     Authorization: Bearer <masked>
I0117 09:47:02.304559      60 round_trippers.go:473]     User-Agent: oc/4.10.0 (linux/amd64) kubernetes/5f40be4
I0117 09:47:02.444811      60 round_trippers.go:574] Response Status: 200 OK in 140 milliseconds
...
---> accessing a build05 image, failed.
$ env "NO_PROXY=*" "no_proxy=*" /cli/oc adm release info  registry.build05.ci.openshift.org/ci-op-kmlwrlwh/release@sha256:a110538b126e0fc74b590497e5e2a0c55aee1699db2f7c5b84b02e7ac5258a87
...
I0117 09:47:53.280922     891 round_trippers.go:463] GET https://registry.build05.ci.openshift.org/openshift/token?scope=repository%3Aci-op-kmlwrlwh%2Frelease%3Apull
I0117 09:47:53.280934     891 round_trippers.go:469] Request Headers:
I0117 09:47:53.280939     891 round_trippers.go:473]     User-Agent: oc/4.10.0 (linux/amd64) kubernetes/5f40be4
I0117 09:47:53.282395     891 round_trippers.go:574] Response Status: 200 OK in 1 milliseconds
I0117 09:47:53.282422     891 round_trippers.go:577] Response Headers:
I0117 09:47:53.282442     891 round_trippers.go:580]     Date: Wed, 17 Jan 2024 09:47:53 GMT
I0117 09:47:53.282450     891 round_trippers.go:580]     Set-Cookie: 64f71fee3e763b226b0debf83db519d1=4bdb552a5e48067e5b0d5f350b9ffbf6; path=/; HttpOnly; Secure; SameSite=None
I0117 09:47:53.282457     891 round_trippers.go:580]     Content-Type: application/json
I0117 09:47:53.282465     891 round_trippers.go:580]     Docker-Distribution-Api-Version: registry/2.0
I0117 09:47:53.282472     891 round_trippers.go:580]     Content-Length: 49
I0117 09:47:53.282581     891 round_trippers.go:463] GET https://registry.build05.ci.openshift.org/v2/ci-op-kmlwrlwh/release/manifests/sha256:a110538b126e0fc74b590497e5e2a0c55aee1699db2f7c5b84b02e7ac5258a87
I0117 09:47:53.282593     891 round_trippers.go:469] Request Headers:
I0117 09:47:53.282612     891 round_trippers.go:473]     Accept: application/vnd.docker.distribution.manifest.list.v2+json
I0117 09:47:53.282622     891 round_trippers.go:473]     Accept: application/vnd.docker.distribution.manifest.v2+json
I0117 09:47:53.282630     891 round_trippers.go:473]     Accept: application/vnd.oci.image.manifest.v1+json
I0117 09:47:53.282639     891 round_trippers.go:473]     Authorization: Bearer <masked>
I0117 09:47:53.282647     891 round_trippers.go:473]     User-Agent: oc/4.10.0 (linux/amd64) kubernetes/5f40be4
I0117 09:47:53.286569     891 round_trippers.go:574] Response Status: 401 Unauthorized in 3 milliseconds
...
```

Reply:
I guess we might hit the same issue in `cucushift-upgrade-toimage` when upgrading to a build05 payload, later we can confirm it.

And `baremetal-upi-ovn-ipv4` is only limited to run in build05 build farm because of network connectivity.

Reply:
get it, sounds like a workaround for build05 and i see the failed job log for cucushift-upgrade-toimage.